CorePluginFramework/src/au/gov/asd/tac/constellation/pluginframework/parameters/PluginParameter.java

```diff
@@ -150,6 +156,38 @@ public void setVisible(boolean visible) {
             fireChangeEvent(ParameterChange.VISIBLE);
         }
     }
+    
+    /**
+     * suppressEvent allows a call to the parameter to stop firing a {@link ParameterChange} event. 
+     * <p>
+     * Specify {@link ParameterChange} event enumerated types as well as a boolean to engage
+     * or disengage them. 
+     * NOTE: passing suppress as false will enable all events.
+     * 
+     * @param suppress if true, the listed events will not fire. if no events are listed, 
+     * all events will not fire. If false, all events are enabled.
+     * @param eventsToSuppress the events to suppress. pass an empty {@link ArrayList}
+     * when specifying all events or enabling events. Pass {@link ParameterChange} 
+     * enumerated types when specifying certain events to suppress.
+     */
+    public void suppressEvent(final boolean suppress, final ArrayList<ParameterChange> eventsToSuppress){
```

Comment:
Please change the parameter type to `List`.

I might be alone in thinking that it feels a little weird to specify an empty list to suppress everything but passing a populated list to ignore specific ones. Perhaps we could have another method being `public void suppressEvent(final boolean suppress)` which will just call this `suppressEvent` with the empty list.

Happy to be told otherwise.

Reply:
passing a populated list should suppress only the ParameterChange events listed within it.
I've done the other changes and pushed them, but yet to change the suppressEvent(...)

Reply:
> passing a populated list should suppress only the ParameterChange events listed within it.

Yep, so if we go with that logic then passing an empty list should suppress nothing right? But it suppresses everything. I'm OK with this though because there is javadoc explaining it. My only minor suggestion is that we could have another method signature of `public void suppressEvent(final boolean suppress)` which will hide the detail of needing to pass the empty list.